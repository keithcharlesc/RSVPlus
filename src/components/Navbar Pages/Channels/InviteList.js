import React from "react";
//Component used in Channels
export default class InviteList extends React.Component {
  constructor() {
    super();
    this.state = {
      inputSize: 0,
    };
  }

  handleOnChange(value) {
    this.setState({
      inputSize: value.target.value,
    });
  }

  renderInputs(value) {
    const inputs = [];
    for (let i = 0; i < value; i++) {
      inputs.push(
        <div key={i}>
          <input type="text" name="emailAddress" placeholder="Gmail Address" />
        </div>
      );
    }
    return inputs;
  }

  render() {
    //console.log(this.state.inputSize);
    return (
      <div>
        <input
          type="tel"
          name="quantity"
          min="1"
          max="9"
          maxLength="1"
          onChange={(value) => this.handleOnChange(value)}
        />
        <div>{this.renderInputs(this.state.inputSize)}</div>
      </div>
    );
  }
}
