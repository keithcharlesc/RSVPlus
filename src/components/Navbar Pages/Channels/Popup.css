.popup {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: auto;
  overflow: auto;
  display: flex;
  margin-top: 60px;
  justify-content: center;
  align-content: center;
  /*background-color: rgba(0, 0, 0, 0.2);*/
}

.popup-inner {
  border-radius: 25px;
  position: relative;
  padding: 32px;
  width: 100vh;
  max-width: 500px;
  background-color: rgb(45, 42, 42);
}

.popup-inner .close-btn {
  position: absolute;
  top: 16px;
  right: 16px;
}
