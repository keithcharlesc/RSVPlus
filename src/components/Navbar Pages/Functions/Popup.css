.popup {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0.2);

  display: flex;
  justify-content: center;
  align-content: center;
}

.popup-inner {
  border-radius: 25px;
  position: relative;
  padding: 32px;
  width: 100vh;
  max-width: 500px;
  background-color: rgb(45, 42, 42);
}

.popup-inner .close-btn {
  position: absolute;
  top: 16px;
  right: 16px;
}
