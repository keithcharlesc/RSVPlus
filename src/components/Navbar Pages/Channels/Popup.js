import React from "react";
import "./Popup.css";
//Used in Channels.js

function Popup(props) {
  return props.trigger ? (
    <div className="backdrop">
      <div className="popup">
        <div className="popup-inner">
          <button className="close-btn" onClick={() => props.setTrigger(false)}>
            {" "}
            X{" "}
          </button>
          {props.children}
        </div>
      </div>
    </div>
  ) : (
    ""
  );
}

export default Popup;
