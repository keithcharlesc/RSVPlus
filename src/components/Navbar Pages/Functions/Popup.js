import React from "react";
import "./Popup.css";

function Popup(props) {
  return props.trigger ? (
    <div className="backdrop">
      <div className="popup">
        <div className="popup-inner">
          <button className="close-btn" onClick={() => props.setTrigger(false)}>
            {" "}
            X{" "}
          </button>
          {props.children}
        </div>
      </div>
    </div>
  ) : (
    ""
  );
}

export default Popup;
