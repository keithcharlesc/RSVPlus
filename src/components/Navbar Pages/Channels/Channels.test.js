import React from "react";
import { render, getByTestId } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { toHaveTextContent } from "@testing-library/jest-dom";
import MockChannels from "./MockChannels";
import Popup from "./Popup";
import InviteList from "./InviteList";

it("renders without crashing", () => {
  const { getByText } = render(<MockChannels />);
  getByText("Loading...");
});

it("Channels matches snapshot", () => {
  const tree = TestRenderer.create(<MockChannels />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const {} = render(<Popup />);
});

it("Popup matches snapshot", () => {
  const tree = TestRenderer.create(<Popup />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const {} = render(<InviteList />);
});

it("InviteList matches snapshot", () => {
  const tree = TestRenderer.create(<InviteList />).toJSON();
  expect(tree).toMatchSnapshot();
});
