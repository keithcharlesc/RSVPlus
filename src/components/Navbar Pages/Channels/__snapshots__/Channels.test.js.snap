// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channels matches snapshot 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="nav-bar-appName text-danger navbar-brand"
        href="/"
      >
        RSVP+
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/channels"
            href="/channels"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Channels
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/calendar"
            href="/calendar"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Calendar
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/faq"
            href="/faq"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            FAQ
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/contact-us"
            href="/contact-us"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Contact Us
          </a>
        </div>
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <span
            className="nav-bar-name navbar-text"
          >
            Signed in as: 
            Bob
            <img
              className="photo ml-2 rounded-circle"
              src="thttp://email.com/image"
            />
          </span>
        </div>
      </div>
      <button
        className="ml-3 btn btn-outline-primary"
        data-testid="button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Log Out
      </button>
    </nav>
    <div>
      
    </div>
  </div>
  <div
    className="p-3 mb-2 bg-dark text-white"
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <h2
      className="page-header text-center mb-4"
    >
      CHANNELS
    </h2>
    <div
      className="container-fluid"
    >
      <div
        className="d-flex align-items-center justify-content-center mb-4 row"
      >
        
        
        <div>
          <div
            className="d-flex align-items-center justify-content-center col"
          >
            <input
              className="mr-3 form-control"
              placeholder="Channel ID"
              required={true}
              style={
                Object {
                  "height": 30,
                  "width": 200,
                }
              }
              type="channelID"
            />
            <button
              className="d-flex align-items-center justify-content-center btn btn-danger"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "height": 30,
                  "width": 60,
                }
              }
              type="button"
            >
              Join
            </button>
          </div>
        </div>
      </div>
      <div
        className="d-flex align-items-center justify-content-center mb-4 row"
      >
        <button
          className="d-flex align-items-center justify-content-center fetch-events-button mr-4 mb-2"
          onClick={[Function]}
          style={
            Object {
              "height": 30,
              "width": 250,
            }
          }
        >
          CREATE A CHANNEL
        </button>
        <button
          className="d-flex align-items-center justify-content-center fetch-events-button mr-4 mb-2"
          onClick={[Function]}
          style={
            Object {
              "height": 30,
              "width": 250,
            }
          }
        >
          DELETE A CHANNEL
        </button>
        <button
          className="d-flex align-items-center justify-content-center fetch-events-button mr-4 mb-2"
          onClick={[Function]}
          style={
            Object {
              "height": 30,
              "width": 250,
            }
          }
        >
          MODIFY INVITES
        </button>
      </div>
      <div
        className="row"
      />
    </div>
  </div>
  
  
  
</div>
`;

exports[`InviteList matches snapshot 1`] = `
<div>
  <input
    className="form-control"
    max="9"
    maxLength="1"
    min="1"
    name="quantity"
    onChange={[Function]}
    placeholder="Input a number from 1 to 9"
    required={true}
    type="tel"
  />
  <div />
</div>
`;

exports[`Popup matches snapshot 1`] = `""`;
